diagnostic.js

```diff
@@ -31,31 +31,91 @@ mongoose.connect('mongodb://localhost/mongoose-crud-diagnostic');
 const Movement = require('./models/movement.js');
 
 /// ADD YOUR CODE BELOW
+const db = mongoose.connection;
+const done = function() {
+  db.close();
+};
 
-const create = (name, description, startYear, endYear) => {};
+const create = (name, description, startYear, endYear) => {
+  Movement.create({
+    name: name,
+    description: description,
+    startYear: startYear,
+    endYear: endYear
+  }).then(function(movement) {
+    console.log(movement.toJSON());
+  }).catch(console.error).then(done);
+};
 // Success -> console.log new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  let search = {};
+  if (arguments[0] && arguments[1]) {
+    let field = arguments[0];
+    let criterion = arguments[1];
+
+    if (criterion[0] === '/') {
+      let regex = new RegExp(criterion.slice(1, criterion.length -1));
+      search[field] = regex;
+    } else {
+      search[field] = criterion;
+    }
+  }
+  Movement.find({}).then(function(movements) {
+    movements.forEach(function(movements) {
+      console.log(movements.toJSON());
+    });
+  }).catch(console.error).then(done);
+};
 // Success -> console.log all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+
+const show = function(id) {
+  Movement.findById(id, function(req,res){
+    if(!res) {
+      .then(console.log("not found"));
+    }
+    else {
+      .then(function(Movement) {
+      console.log(person.toJSON());
+    })};
+  ).catch(console.error).then(done);
+    };
```

Comment:
it seems your chaining `then` onto nothing. Also we weren't looking for a request or response here